src/brski/pledge/pledge_request.cpp

```diff
@@ -192,14 +192,25 @@ std::string create_cert_string(const char *cert) {
   return out;
 }
 
+uint64_t gen_rand64(void) {
+  uint64_t value = 0x0;
+
+  srand(time(0));
+
+  for (int i = 0; i < 64; i += 15) {
+    value = value * ((uint64_t)RAND_MAX + 1) + rand();
+  }
+  return value;
+}
```

Comment:
Btw, C++11 has a built-in random number generator you can use instead, see [`<random>`](https://en.cppreference.com/w/cpp/numeric/random)

```suggestion
#include <random>

uint64_t gen_rand64(void) {
    static auto engine = std::default_random_engine();
    static auto generator = std::uniform_int_distribution<uint64_t>();
    return generator(engine);
}
```

Also, if you're adding a function only in a `.cpp`/`.c` file and not in a `.h` file, you should also make sure to declare the function as `static` (works in C++ and C) or to put it into an anonymous namespace (C++ only).